src/libraries/System.Text.Json/tests/Common/PropertyVisibilityTests.InitOnly.cs

```diff
@@ -49,6 +49,18 @@ public virtual async Task NonPublicInitOnlySetter_With_JsonInclude(Type type)
             Assert.Equal(@"{""MyInt"":1}", await Serializer.SerializeWrapper(obj));
         }
 
+        [Theory]
+        [InlineData(typeof(ClassWithPublic_InitOnlyProperty_WithJsonIgnoreProperty))]
+        public async Task PublicInitOnlySetter_With_JsonIgnore(Type type)
```

Comment:
Can you also add tests for these polymorphic scenarios (https://dotnetfiddle.net/caqjtK), here and in the unit tests? Ensure that no warnings are emitted for ignored properties.

```cs
using System;
using System.Text.Json;
using System.Text.Json.Serialization;
					
public class Program
{
	public static void Main()
	{
		string json = JsonSerializer.Serialize(new Derived1());
		Console.WriteLine(json);
		
		json = JsonSerializer.Serialize(new Derived2());
		Console.WriteLine(json);
	}
	
	public class Base1
	{
		public int MyInt { get; init; } = 3;
	}
	
	public class Derived1 : Base1
	{
		[JsonIgnore]
		public new int MyInt { get; init; } = 4;
	}
	
	public class Base2
	{
		public virtual int MyInt { get; init; } = 3;
	}
	
	public class Derived2 : Base2
	{
		[JsonIgnore]
		public override int MyInt { get; init; } = 4;
	}
}
```

Reply:
There's a bunch of reliability issues concerning the semantics of JsonIgnoreAttribute, see #59675, #50078, #66900, #60082. I think we might want to consider ironing these out before merging this PR. This might likely happen after .NET 7.